### PR TITLE
[autoparallel] integrate_gpt_related_tests

### PR DESCRIPTION
**Pull request by YuliangLiu0306**
### What does this PR do
1. This PR integrates all gpt related tests into a folder named 'test_gpt'. 
2. All the solver strategy tests are integrated into test_solver_with_gpt_modules.py.
3. All the runtime numerical tests  are integrated into test_runtime_with_gpt_modules.py.
4. All the gpt modules which are reused in both test files are integrated into gpt_modules.py.

### CODE REVIEW

File: tests/test_auto_parallel/test_tensor_shard/test_gpt/gpt_modules.py [hunk 1]
```diff
@@ -125,15 +123,11 @@ def forward(
         present = (key, value)
 
         attn_output, attn_weights = self._attn(query, key, value, attention_mask, head_mask)
-
+        attn_output = attn_output
```
Comment (feifeibear) [inline tests/test_auto_parallel/test_tensor_shard/test_gpt/gpt_modules.py:126]: why we need this line

Comment (YuliangLiu0306) [reply tests/test_auto_parallel/test_tensor_shard/test_gpt/gpt_modules.py:126]: This line could be removed.

Comment (YuliangLiu0306) [reply tests/test_auto_parallel/test_tensor_shard/test_gpt/gpt_modules.py]: Done.

File: tests/test_auto_parallel/test_tensor_shard/test_gpt/gpt_modules.py [hunk 2]
```diff
@@ -2,32 +2,30 @@
...
+
+    def __init__(self, intermediate_size, config):
+        super().__init__()
+        embed_dim = config.hidden_size
+        self.c_fc = Conv1D(intermediate_size, embed_dim)
+        self.c_proj = Conv1D(embed_dim, intermediate_size)
+        self.act = ACT2FN[config.activation_function]
+        # We temporarily banned the Dropout layer because the rng state need
+        # to process to get the correct result.
+        # self.dropout = nn.Dropout(config.resid_pdrop)
```
Comment (feifeibear) [inline tests/test_auto_parallel/test_tensor_shard/test_gpt/gpt_modules.py:20]: you can use model.eval() during test. eval will ban dropout

Comment (YuliangLiu0306) [reply tests/test_auto_parallel/test_tensor_shard/test_gpt/gpt_modules.py:20]: okay

Comment (YuliangLiu0306) [reply tests/test_auto_parallel/test_tensor_shard/test_gpt/gpt_modules.py:20]: I tried model.eval(), it not only ban the dropout module, but change the behavior of LayerNorm module as well. I think comment that line manually is a safer way in unit test.

**Review by FrankLeeeee — APPROVED** [review]

File: tests/test_auto_parallel/test_tensor_shard/test_gpt/gpt_modules.py (hunk 2 above)
Comment (FrankLeeeee) [reply tests/test_auto_parallel/test_tensor_shard/test_gpt/gpt_modules.py:20]: This raises another question for our consideration even though it is definitely not supposed to be handled in this PR. PyTorch users often do evaluation after training for a few epochs, I don't think we can support this mode change for now, but good to keep it in mind.
